group-coordinator/src/main/java/org/apache/kafka/coordinator/group/streams/topics/EndpointToPartitionsManager.java

```diff
@@ -47,11 +46,11 @@ public static StreamsGroupHeartbeatResponseData.EndpointToPartitions endpointToP
 
             final Map<String, TopicMetadata> groupTopicMetadata = streamsGroup.partitionMetadata();
             Set<Map.Entry<String, Set<Integer>>> taskEntrySet = streamsGroupMember.assignedActiveTasks().entrySet();
-            List<StreamsGroupHeartbeatResponseData.TopicPartition> topicPartitionList = getTopicPartitions(taskEntrySet, sourceTopics, repartitionSourceTopics, groupTopicMetadata);
+            List<StreamsGroupHeartbeatResponseData.TopicPartition> topicPartitionList = getTopicPartitions(taskEntrySet, sourceTopics, groupTopicMetadata);
```

Comment:
I was incorrectly combining source and repartition topics - they should be added separately since we are sending back standby tasks separately

Reply:
It's good that you wrote an integration test, so I can trust that you are doing the right thing here. I just don't really understand why IQv2 works this way. Why do we only send input topics for active taskss and repartition topics for standby tasks?

Reply:
@lucasbru - good question.  I'll update the test to be more clear, but it should be the case that active tasks for both instances get source partitions.  Since there's two sub-topologies with two partitions, each instance gets one of each.